### PR TITLE
refactor: cred-def controller APIs

### PR DESCRIPTION
**Pull request by bhavanakarwade**
### What?
- Modify GET API of `/orgs/{orgId}/cred-defs/{credDefId}` (get an existing cred def by id).
- Modify GET API of `/verification/cred-defs/{schemaId}` (get an existing cred def by schema id).
- Modify GET API of `/orgs/{orgId}/cred-defs` (get all cred-defs of provided org id with pagination).
- Modify GET API of `/orgs/{orgId}/bulk/cred-defs` (fetch all cred-defs for bulk operation).
- Modify  POST API of `/orgs/{orgId}/cred-defs` (create and sends a cred-def to the ledger).

1.Checked validations on payload.
2.Refactor response
3.Ensured consistency of API status codes.
4.Modified the NATS function.

### Why?
- The implementation of a well-defined interface enhances code readability and makes the structure of the API response more explicit.
- Ensuring consistent API status codes improves communication between different components of the system and enhances overall reliability.

### How?
Replaced the return type with a well-defined interface to enhance code clarity.
Checked and standardized API status codes to ensure a uniform response.
Modified the NATS function to accommodate the adjustments made in the API.

### CODE REVIEW

**Review by pranalidhanavade — APPROVED** [review]
LG

File: apps/api-gateway/src/credential-definition/credential-definition.controller.ts [hunk 1]
```diff
@@ -38,15 +38,15 @@ export class CredentialDefinitionController {
   @Roles(OrgRoles.OWNER, OrgRoles.ADMIN, OrgRoles.ISSUER, OrgRoles.VERIFIER, OrgRoles.MEMBER)
   @UseGuards(AuthGuard('jwt'), OrgRolesGuard)
   async getCredentialDefinitionById(
-    @Param('orgId') orgId: string,
+    @Param('orgId', new ParseUUIDPipe({exceptionFactory: (): Error => { throw new BadRequestException(`Invalid format for orgId`); }})) orgId: string,
```
Comment (tipusinghaw) [inline apps/api-gateway/src/credential-definition/credential-definition.controller.ts:41]: Please add this message `Invalid format for orgId` in a common file

File: apps/api-gateway/src/credential-definition/credential-definition.controller.ts [hunk 2]
```diff
@@ -38,15 +38,15 @@ export class CredentialDefinitionController {
   @Roles(OrgRoles.OWNER, OrgRoles.ADMIN, OrgRoles.ISSUER, OrgRoles.VERIFIER, OrgRoles.MEMBER)
```
Comment (tipusinghaw) [inline apps/api-gateway/src/credential-definition/credential-definition.controller.ts:38]: Summary and description should not be same

File: apps/api-gateway/src/credential-definition/credential-definition.controller.ts [hunk 3]
```diff
@@ -61,12 +61,13 @@ export class CredentialDefinitionController {
   async getCredentialDefinitionBySchemaId(
```
Comment (tipusinghaw) [inline apps/api-gateway/src/credential-definition/credential-definition.controller.ts:61]: Summary and description should not be same

File: apps/api-gateway/src/credential-definition/credential-definition.controller.ts [hunk 4]
```diff
@@ -76,45 +77,48 @@ export class CredentialDefinitionController {
     summary: 'Fetch all credential definitions of provided organization id with pagination',
```
Comment (tipusinghaw) [inline apps/api-gateway/src/credential-definition/credential-definition.controller.ts:77]: I think this message is misleading

File: apps/api-gateway/src/credential-definition/credential-definition.controller.ts [hunk 5]
```diff
@@ -76,45 +77,48 @@ export class CredentialDefinitionController {
...
       message: ResponseMessages.credentialDefinition.success.fetch,
-      data: credentialsDefinitionDetails.response
+      data: credentialsDefinitionDetails
     };
     return res.status(HttpStatus.OK).json(credDefResponse);
   }
 
+  @Get('/orgs/:orgId/bulk/cred-defs')
+  @ApiOperation({
+    summary: 'Fetch all credential definition for bulk opeartion',
```
Comment (tipusinghaw) [inline apps/api-gateway/src/credential-definition/credential-definition.controller.ts:103]: I think this message is misleading, Please check the message content

File: apps/api-gateway/src/credential-definition/credential-definition.controller.ts [hunk 6]
```diff
@@ -76,45 +77,48 @@ export class CredentialDefinitionController {
...
+    description: 'Fetch all credential definition from metadata saved in database for bulk opeartion.'
+  })
+  @ApiResponse({ status: HttpStatus.OK, description: 'Success', type: ApiResponseDto })
+  @Roles(OrgRoles.OWNER, OrgRoles.ADMIN, OrgRoles.ISSUER, OrgRoles.VERIFIER)
+  @UseGuards(AuthGuard('jwt'), OrgRolesGuard)
+  async getAllCredDefAndSchemaForBulkOperation(
+    @Param('orgId', new ParseUUIDPipe({exceptionFactory: (): Error => { throw new BadRequestException(`Invalid format for orgId`); }})) orgId: string,
```
Comment (tipusinghaw) [inline apps/api-gateway/src/credential-definition/credential-definition.controller.ts:110]: Add this message in a common file "Invalid format for orgId".

File: apps/ledger/src/credential-definition/repositories/credential-definition.repository.ts [hunk 7]
```diff
@@ -5,6 +5,8 @@ import { credential_definition, org_agents, org_agents_type, organisation, schem
 import { Injectable, Logger } from '@nestjs/common';
 import { ResponseMessages } from '@credebl/common/response-messages';
 import { BulkCredDefSchema, CredDefSchema } from '../interfaces/credential-definition.interface';
+import { ICredDefData } from '@credebl/common/interfaces/cred-def.interface';
+import { SortValue, sortValue } from 'apps/api-gateway/src/enum';
```
Comment (tipusinghaw) [inline apps/ledger/src/credential-definition/repositories/credential-definition.repository.ts:9]: Please create this `enum` in libs folder

File: apps/ledger/src/credential-definition/credential-definition.service.ts [hunk 8]
```diff
@@ -17,6 +17,8 @@ import { map } from 'rxjs/operators';
 import { OrgAgentType } from '@credebl/enum/enum';
 import { Cache } from 'cache-manager';
 import { CACHE_MANAGER } from '@nestjs/cache-manager';
+import { ICredDefDetails } from '@credebl/common/interfaces/cred-def.interface';
+import { sortValue } from 'apps/api-gateway/src/enum';
```
Comment (tipusinghaw) [inline apps/ledger/src/credential-definition/credential-definition.service.ts:21]: We should add the gereric `enums` in libs folder

File: apps/api-gateway/src/credential-definition/credential-definition.controller.ts [hunk 9]
```diff
@@ -76,45 +77,48 @@ export class CredentialDefinitionController {
     summary: 'Fetch all credential definitions of provided organization id with pagination',
     description: 'Fetch all credential definitions from metadata saved in database of provided organization id.'
```
Comment (tipusinghaw) [inline apps/api-gateway/src/credential-definition/credential-definition.controller.ts:78]: I think this message `from metadata` is misleading.

File: apps/api-gateway/src/credential-definition/credential-definition.controller.ts [hunk 10]
```diff
@@ -76,45 +77,48 @@ export class CredentialDefinitionController {
...
-      data: credentialsDefinitionDetails.response
+      data: credentialsDefinitionDetails
     };
     return res.status(HttpStatus.OK).json(credDefResponse);
   }
 
+  @Get('/orgs/:orgId/bulk/cred-defs')
+  @ApiOperation({
+    summary: 'Fetch all credential definition for bulk opeartion',
+    description: 'Fetch all credential definition from metadata saved in database for bulk opeartion.'
```
Comment (tipusinghaw) [inline apps/api-gateway/src/credential-definition/credential-definition.controller.ts:104]: I think this message is misleading, Please check the message content

File: apps/api-gateway/src/credential-definition/credential-definition.controller.ts (hunk 1 above)
Comment (bhavanakarwade) [reply apps/api-gateway/src/credential-definition/credential-definition.controller.ts:41]: Resolved

File: apps/api-gateway/src/credential-definition/credential-definition.controller.ts (hunk 6 above)
Comment (bhavanakarwade) [reply apps/api-gateway/src/credential-definition/credential-definition.controller.ts:110]: Resolved

File: apps/api-gateway/src/credential-definition/credential-definition.controller.ts (hunk 9 above)
Comment (bhavanakarwade) [reply apps/api-gateway/src/credential-definition/credential-definition.controller.ts:78]: Resolved

File: apps/api-gateway/src/credential-definition/credential-definition.controller.ts (hunk 10 above)
Comment (bhavanakarwade) [reply apps/api-gateway/src/credential-definition/credential-definition.controller.ts:104]: Resolved

File: apps/api-gateway/src/credential-definition/credential-definition.controller.ts (hunk 2 above)
Comment (bhavanakarwade) [reply apps/api-gateway/src/credential-definition/credential-definition.controller.ts:38]: Resolved

File: apps/api-gateway/src/credential-definition/credential-definition.controller.ts (hunk 3 above)
Comment (bhavanakarwade) [reply apps/api-gateway/src/credential-definition/credential-definition.controller.ts:61]: Resolved

File: apps/ledger/src/credential-definition/repositories/credential-definition.repository.ts (hunk 7 above)
Comment (bhavanakarwade) [reply apps/ledger/src/credential-definition/repositories/credential-definition.repository.ts:9]: Resolved

File: apps/ledger/src/credential-definition/credential-definition.service.ts (hunk 8 above)
Comment (bhavanakarwade) [reply apps/ledger/src/credential-definition/credential-definition.service.ts:21]: Resolved

**Review by tipusinghaw — APPROVED** [review]
LGTM

**Pull request merged**